packages/@lwc/shared/src/api-version.ts

```diff
@@ -125,5 +130,7 @@ export function isAPIFeatureEnabled(
         case APIFeature.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE:
         case APIFeature.USE_LIGHT_DOM_SLOT_FORWARDING:
             return apiVersion >= APIVersion.V61_250_SUMMER_24;
+        case APIFeature.ENABLE_COMPONENT_CLASS_TRANSFORM_FOR_HMR_HOOKS:
+            return apiVersion >= APIVersion.V62_252_WINTER_25;
```

Comment:
I don't think this needs to go behind API versioning. It is exceedingly unlikely that a customer already has a variable named `__lwc_component_class_internal`.